client/my-sites/site-settings/form-general.jsx

```diff
@@ -601,6 +601,10 @@ export class SiteSettingsFormGeneral extends Component {
 		} );
 		const isDevelopmentSite = Boolean( site?.is_a4a_dev_site );
 
+		if ( ! site ) {
+			return null;
+		}
```

Comment:
If anyone knows why this is needed I'd like to know. This "fixes" the problem but I think it's just masking the root cause, why is both site being unset after navigation and the page still trying to render this form?

Reply:
I may not be answering the question, but I remember that navigating to /sites triggers a site state reset? Adding a guard for `site=null` could be beneficial in general, similar to the approach used at https://github.com/Automattic/wp-calypso/pull/95981/files#diff-bc79a829db32a21a70056887cf8a6bd4ebce6890c1a8fd3091258dba0610b680R610